autofit/graphical/expectation_propagation/factor_optimiser.py

```diff
@@ -16,11 +16,12 @@ class AbstractFactorOptimiser(ABC):
     """
     logger = logger.debug
 
-    def __init__(self, initial_values=None, deltas=None, inplace=False, delta=1):
+    def __init__(self, initial_values=None, deltas=None, inplace=False, delta=1, dynamic_delta=False):
```

Comment:
Currently the default optimiser would need to be passed with dynamic_delta=True explicitly. I'm tempted to make the default dynamic_delta=True

Reply:
Once I test this feature set properly I suspect we'll end up making it True by default.

Reply:
Yeah makes sense. I also need to look at incorporating it into the Laplace optimiser

Reply:
Oh wait that's what this PR does lol